src/components/textarea/TextArea.spec.tsx

```diff
@@ -0,0 +1,187 @@
+        let wrapper: ReactWrapper<ITextAreaProps, any>;
+        let textArea: ReactWrapper<HTMLAttributes, any>;
+
+        beforeEach(() => {
+            wrapper = mount(
+                <TextArea id='textarea-id' />,
+                {attachTo: document.getElementById('App')},
+            );
+
+            textArea = wrapper.find('textarea').first();
+        });
+
+        afterEach(() => {
+            wrapper.unmount();
+            wrapper.detach();
+        });
+
+        it('should set textarea id when specified', () => {
+            expect(textArea.prop('id')).toBe('textarea-id');
+        });
+
+        it('should set className when specified', () => {
+            const className = 'a-class';
+            expect(textArea.hasClass(className)).toBe(false);
+
+            wrapper.setProps({className}).mount();
+            expect(textArea.hasClass(className)).toBe(true);
+        });
+
+        it('should set additionalAttributes when specified', () => {
+            expect(textArea.prop('placeholder')).toBeUndefined();
+            wrapper.setProps({additionalAttributes: {placeholder: 'not null'}}).mount();
+            expect(textArea.prop('placeholder')).toBe('not null');
+        });
+
+        it('should set disabled prop when specified', () => {
+            expect(textArea.prop('disabled')).toBeUndefined();
+            wrapper.setProps({disabled: true}).mount();
+            expect(textArea.prop('disabled')).toBe(true);
+        });
+
+        it('should set value prop when specified', () => {
+            expect(textArea.prop('value')).toBeUndefined();
+            wrapper.setProps({value: 'non empty'}).mount();
+            expect(textArea.prop('value')).toBe('non empty');
+        });
+
+        it('should call prop onChange on textarea change', () => {
+            const onChange = jasmine.createSpy('onChange');
+
+            wrapper.setProps({onChange}).mount();
+            textArea.simulate('change');
+
+            expect(onChange).toHaveBeenCalledTimes(1);
+        });
+
+        it('should call prop onMount on mount', () => {
+            const onMount = jasmine.createSpy('onMount');
+
+            wrapper.setProps({onMount}).unmount().mount();
+
+            expect(onMount).toHaveBeenCalledTimes(1);
+        });
+
+        it('should call prop onUnmount on Unmount', () => {
+            const onUnmount = jasmine.createSpy('onUnmount');
+
+            wrapper.setProps({onUnmount}).unmount();
+
+            expect(onUnmount).toHaveBeenCalledTimes(1);
+        });
+
+        describe('<TextAreaConnected />', () => {
+            let store: Store<IReactVaporState>;
+            let textAreaProps: ITextAreaProps;
+
+            const mountComponentWithProps = (props: ITextAreaProps) => mount(
+                <Provider store={store}>
+                    <TextAreaConnected {...props} />
+                </Provider>,
+                {attachTo: document.getElementById('App')},
+            );
+

```

Comment:
Since you often use the same logic with findWhere, you could extract this in a helper function:
```typescript
const getTextAreaStateFromId = (id: string): ITextAreaState => findWhere(store.getState().textAreas, {id});
```
And use it in your tests like so:
```typescript
getTextAreaStateFromId(textAreaProps.id);
```

Reply:
done